service/policy/db/namespaces.go

```diff
@@ -298,19 +175,20 @@ func (c PolicyDBClient) UnsafeUpdateNamespace(ctx context.Context, id string, na
 		}
 	}
 
-	return hydrateNamespaceItem(row, namespaceSelectOptions{}, c.logger)
-}
+	metadata := &common.Metadata{}
+	if err = unmarshalMetadata(updatedNs.Metadata, metadata, c.logger); err != nil {
+		return nil, err
+	}
 
-func setNamespaceActiveStateSQL(id string, isActive bool) (string, []interface{}, error) {
-	t := Tables.Namespaces
-	return db.NewStatementBuilder().
-		Update(t.Name()).
-		Set("active", isActive).
-		Where(sq.Eq{t.Field("id"): id}).
-		Suffix("RETURNING \"id\"").
-		ToSql()
+	return &policy.Namespace{
+		Id:       updatedNs.ID,
+		Name:     updatedNs.Name,
+		Active:   &wrapperspb.BoolValue{Value: updatedNs.Active},
+		Metadata: metadata,
+	}, nil
 }
 
+// todo: should be renamed to UnsafeDeactivateNamespace for consistency
```

Comment:
For discussion, should this method have an `Unsafe` prefix like the rest?

Reply:
Good question. Deactivations are safe, and reactivations are unsafe because they can lead to new access to existing data: https://github.com/opentdf/platform/issues/108

Reply:
Comment removed